userspace/libsinsp/user.cpp

```diff
@@ -32,7 +32,88 @@ limitations under the License.
 #endif
 
 #if defined(HAVE_PWD_H) || defined(HAVE_GRP_H)
+
+//
+// NOTE: at the time of writing 16 would have been enough, being
+// the format `mnt:[<unsigned>]`, but the only call to `ns_get_name`
+// I could find in the kernel was using a buffer of 50, so 50 it is.
+//
+#define NS_MNT_SIZE 50
+
+#include <unistd.h>
+
 std::string sinsp_usergroup_manager::s_host_root;
+
+// See fgetpwent() / fgetgrent() feature test macros:
+// https://man7.org/linux/man-pages/man3/fgetpwent.3.html
+// https://man7.org/linux/man-pages/man3/fgetgrent.3.html
+#if defined _DEFAULT_SOURCE || defined _SVID_SOURCE
+#define HAVE_FGET__ENT
+#endif
+
+const char *get_root_ns_mnt()
+{
+	static char *s_root_ns_mnt = nullptr;
```

Comment:
Can we make this a class field? And free it in the dtor?
Same for `CANNOT_READ_ROOT_NS_MNT`.
Perhaps we can load them in the ctor.

Reply:
In its own class!